### PR TITLE
Homework1-needs-some-refactoring

### PR DESCRIPTION
**Pull request by Marji9988**

### CODE REVIEW

**Comment by Marji9988** [comment]
Would love all the feedback/remarks I can get.

File: routes/userAccount.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+router.route('/')
+  .get(readGeneralUserAccount)
+  .post(createNewAccount)
+  .put(editUserAccount);
+
+router.route('/accounts/:id')
+  .get(getUserAccount)
+  .delete(deleteAccount);
+
+router.route('/income')
```
Comment (AudriusBaltusis) [inline routes/userAccount.js:22]: This and following routes are not related to `userAccount`. You should use more generic module name, or split to different modules

File: controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    currency,
+    balance,
+    id: uuidv4(),
+  };
+  userAccount.accounts.push(newAccount);
+  res.json(newAccount);
+};
+
+// Creates new income and adds it up to the user account income
+const createNewIncome = (req, res) => {
```
Comment (AudriusBaltusis) [inline controllers/user.js:78]: This and some other methods are not related to `user` and should be moved to another module

File: controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+// making all the expenses 0. (will be refactored for sure!)
+// const deleteExpenses = (req, res) => {
+//   const { expenses } = userAccount;
+//   expenses.all = 0;
+//   res.json(expenses);
+// };
+
+const deleteExpenseById = (req, res) => {
+  const { id } = req.params;
+  // eslint-disable-next-line max-len
```
Comment (AudriusBaltusis) [inline controllers/user.js:153]: Please adjust your `eslint` settings and reformat code

File: controllers/user.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+// Deleting one of the user accounts by id
+const deleteAccount = (req, res) => {
+  const { id } = req.params;
+  const accountToDelete = userAccount.accounts.find((account) => account.id.toString() === id);
+  // eslint-disable-next-line max-len
+  userAccount.accounts = userAccount.accounts.filter((account) => account.id.toString() !== id.toString());
+  res.json(accountToDelete);
+};
+
+// making all the expenses 0. (will be refactored for sure!)
```
Comment (AudriusBaltusis) [inline controllers/user.js:144]: Please remove unused methods instead of commenting out

File: controllers/user.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+};
+
+// Reads General user expenses (all) (can be refactored)
+const readUserExpenses = (req, res) => {
+  const generalExpenses = userAccount.expenses[0].all;
+  // console.log(typeof userAccount.expenses[0].all);
+  res.json(generalExpenses);
+};
+
+// Creates new account for the user with name, currency, balance and id.
```
Comment (AudriusBaltusis) [inline controllers/user.js:64]: Minor: such comments are redundant, method name is self-explanatory

**Review by AudriusBaltusis — CHANGES_REQUESTED** [review]

**Comment by AudriusBaltusis** [comment]
Any updates on this?

**Comment by Marji9988** [comment]
> Any updates on this?

Hello Audrius. The changes you've requested will be done today in a new pull request, along with the new homework.

**Pull request closed without merging**